### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by MuskanChhatrasal**
Done with Authentication

### CODE REVIEW

File: src/Pages/Authentication/login.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ const Login = () => {
       style={{ marginLeft: "30rem", marginTop: "7rem" }}
     >
       <h2 style={{ fontSize: "2rem" }}>Login</h2>
```
Comment (MayankKumar10) [inline src/Pages/Authentication/login.js:27]: try to avoid inline styling

File: src/Pages/Authentication/login.js [hunk 2]
```diff
@@ -51,17 +51,28 @@ const Login = () => {
...
+              value="Test Login"
+              className="button btn-primary"
+              onClick={() => testlogin()}
+            />
+          </div>
         </div>
-        <div className="text">
+        <div
+          className="text"
+          style={{ marginTop: "2rem", marginLeft: "-1rem" }}
```
Comment (MayankKumar10) [inline src/Pages/Authentication/login.js:74]: try to avoid inline styling

File: src/Pages/Authentication/login.js [hunk 3]
```diff
@@ -51,17 +51,28 @@ const Login = () => {
...
+              className="button btn-primary"
+              onClick={() => submitHandler()}
+            />
+          </div>
+          <div>
+            <input
+              type="Submit"
+              value="Test Login"
+              className="button btn-primary"
+              onClick={() => testlogin()}
```
Comment (MayankKumar10) [inline src/Pages/Authentication/login.js:68]: you don't need to write arrow function to call the function you call with its name only without ( () ) parentheses.

**Review by MayankKumar10 — APPROVED** [review]
Look into these suggestions everything else looks good to me you can merge it.

File: src/Context/authContext.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
+import { createContext, useContext, useReducer } from "react";
+import axios from "axios";
+
+export const InitialAuthData = {
+  authData: {},
+  authErrorMsg: "",
+};
+
+export const AuthReducer = (state, { type, payload }) => {
```
Comment (aayushjain-code) [inline src/Context/authContext.js:9]: you can have a different file for reducers combined

File: src/Pages/Authentication/login.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    } else {
+      console.log(userDetails);
+      login(userDetails);
+      setUserDetails({ email: "", password: "" });
+    }
+  };
+  return (
+    <div
+      className="wrapper login-wrapper"
+      style={{ marginLeft: "30rem", marginTop: "7rem" }}
```
Comment (aayushjain-code) [inline src/Pages/Authentication/login.js:25]: avoid using inline styles

**Review by aayushjain-code — APPROVED** [review]
Nice UI can merge with a few recommendations...

File: src/Components/Navbar/navbar.js [hunk 6]
```diff
@@ -17,36 +19,42 @@ const Navbar = () => {
...
-            class="fas fa-search"
-            style={{
-              color: "white",
-              position: "absolute",
-              marginLeft: "-2rem",
-            }}
-          ></i>
-        </li> */}
-
+        <li style={{ marginTop: "0.75rem", cursor: "pointer", color: "white" }}>
```
Comment (Developers-codz) [inline src/Components/Navbar/navbar.js:22]: avoid inline css

File: src/Pages/Authentication/login.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  const [error, setError] = useState({ isError: true, text: "" });
+  const { login, testlogin } = useAuth();
+
+  const submitHandler = () => {
+    if (!userDetails.email || !userDetails.password) {
+      setError({ isError: true, text: "Please enter all the fields." });
+    } else if (!userDetails.email.includes("@")) {
+      setError({ isError: true, text: "Invalid Email" });
+    } else {
+      console.log(userDetails);
```
Comment (Developers-codz) [inline src/Pages/Authentication/login.js:17]: remove console

File: src/Pages/Authentication/signup.js [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+      setError({ isError: true, text: "Please enter all the fields" });
+    } else if (userDetails.password !== userDetails.confirmPassword) {
+      setError({
+        isError: true,
+        text: "Password and confirm password does not match",
+      });
+    } else if (!userDetails.email.includes("@")) {
+      setError({ isError: true, text: "Invalid email" });
+    } else {
+      console.log(userDetails);
```
Comment (Developers-codz) [inline src/Pages/Authentication/signup.js:37]: here also

File: src/Pages/Home/home.js [hunk 9]
```diff
@@ -7,10 +7,18 @@ import VideoCard from "../../Components/VideoCard/videoCard";
...
 
-  useEffect(async () => {
-    const response = await axios.get("/api/videos");
-    console.log(response.data.videos);
-    setVideos(response.data.videos);
+  // useEffect(async () => {
+  //   const response = await axios.get("/api/videos");
+  //   console.log(response.data.videos);
+  //   setVideos(response.data.videos);
+  // }, []);
```
Comment (Developers-codz) [inline src/Pages/Home/home.js:14]: remove commented code

**Review by Developers-codz — APPROVED** [review]
Everything looks good to merge
Just added few observation, also for horizontal scroll, you can try  giving 100vw to body tag to remove scroll

File: src/Pages/Home/home.js (hunk 9 above)
Comment (MuskanChhatrasal) [reply src/Pages/Home/home.js:14]: Thankyou for your suggestions. Will surely try to implement it.

File: src/Context/authContext.js (hunk 4 above)
Comment (MuskanChhatrasal) [reply src/Context/authContext.js:9]: Thankyou. Will surely implementing it.

File: src/Pages/Authentication/login.js (hunk 3 above)
Comment (MuskanChhatrasal) [reply src/Pages/Authentication/login.js:68]: Thankyou. Will do that.

**Pull request merged**